tests/test_boundary_trie.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+
+from modguard.core import BoundaryTrie, PublicMember, BoundaryNode
+
+
+@pytest.fixture
+def boundary_trie():
+    return BoundaryTrie(
+        root=BoundaryNode(
+            public_members={},
+            children={
+                "domain_one": BoundaryNode(
+                    public_members={},
+                    children={},
+                    is_end_of_path=True,
+                    full_path="domain_one",
+                ),
+                "domain_two": BoundaryNode(
+                    public_members={},
+                    children={
+                        "subdomain": BoundaryNode(
+                            public_members={},
+                            children={},
+                            is_end_of_path=True,
+                            full_path="domain_two.subdomain",
+                        )
+                    },
+                    is_end_of_path=True,
+                    full_path="domain_two",
+                ),
+                "domain_three": BoundaryNode(
+                    public_members={},
+                    children={},
+                    is_end_of_path=True,
+                    full_path="domain_three",
+                ),
+                "domain_four": BoundaryNode(
+                    public_members={
+                        "domain_four.public_api": PublicMember(
+                            name="domain_four.public_api", allowlist=None
+                        )
+                    },
+                    children={},
+                    is_end_of_path=True,
+                    full_path="domain_four",
+                ),
+            },
+            is_end_of_path=True,
+            full_path="",
+        )
+    )
+
+
+def test_iterate_over_empty_trie():
+    assert list(BoundaryTrie()) == []
+
+
+def test_iterate_over_populated_trie(boundary_trie):
+    assert set((node.full_path for node in boundary_trie)) == {
+        "",
+        "domain_one",
+        "domain_two",
+        "domain_two.subdomain",
+        "domain_three",
+        "domain_four",
+    }
+
+
+def test_get_nonexistent_path(boundary_trie):
+    assert boundary_trie.get("fakepath") is None
+
+
+def test_get_nonexistent_empty_path():
+    trie = BoundaryTrie()
+    assert trie.get("") is None
+
+
+def test_get_actual_path(boundary_trie):
+    assert boundary_trie.get("domain_one") is not None
+
+
+def test_get_actual_empty_path(boundary_trie):
+    assert boundary_trie.get("") is not None
```

Comment:
could be nice to have a stricter assertion than `is not None` ie check against the expected returned value